Islandora/islandora_scholar
# Adds check to OpenURL PID in admin form

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1923

# What does this Pull Request do?
Adds a check to validate the user-supplied email address used as the OpenURL PID.

# What's new?
Theres a new function in islandora_doi/includes/utilities.inc, `islandora_doi_openurl_pid_valid()`, that takes a supplied email and uses it to perform a CrossRef request on an example DOI (fake DOI thats valid according to CrossRef used for testing) and judges whether its valid based on the content-type of CrossRef's response. If the email address is registered with CrossRef, they get an XML response. If the email is not registered with CrossRef, they get an HTML error message. 

The admin form feeds the OpenURL PID supplied by the user to this function during a validation step; if `islandora_doi_openurl_pid_valid()` determines that the email address is invalid, users get an error message and the invalid email is not saved.

# How should this be tested?
On a fresh VM, go to the admin menu for Scholar and go to the "DOI Configuration" tab. Try to enter a random email address other than the default that you know isn't registered with CrossRef. It shouldn't let you do it and will give you an error message asking you if the provided email has been registered. If you register an email with CrossRef (or you can just use "bjbrown@fsu.edu" if you don't want to go through the hassle) and provide it, it should allow you to save that valid email address.

# Interested parties
@Islandora/7-x-1-x-committers
